discovery-provider/src/model_validator.py

```diff
@@ -0,0 +1,136 @@
+import json
+import copy
+import logging  # pylint: disable=C0302
+from jsonschema import Draft7Validator, ValidationError
+logger = logging.getLogger(__name__)
+
+# https://app.quicktype.io/ -- JSON schema generator
+class ModelValidator:
+    '''
+        {key: value} :
+        {
+            model (first letter uppercase): {
+                model_schema: schema,
+                field_schema: {
+                    title: schema,
+                    mood: schema,
+                    ...
+            },
+                fields: [title, mood, ...]
+            }
+        }
+    '''
+    models_to_schema_and_fields_dict = {}
+
+    @classmethod
+    def validate(cls, to_validate, model, field=""):
+        schema = cls.get_schema(field, model)
+        validator = Draft7Validator(schema)
+
+        try:
+            found_invalid_field = False
+            for error in sorted(validator.iter_errors(to_validate), key=str):
+                found_invalid_field = True
+                logger.warning('Error with {0} instance {1}: {2}'.format(model, to_validate, error.message))
+
+            # if any error occurs, raise exception
+            if found_invalid_field:
+                raise ValidationError('Instance {0} is not proper'.format(to_validate))
+        except ValidationError as e:
+            raise e
+
+    @classmethod
+    def get_schema(cls, field, model):
+        # If model is not present in dict, init its schema and its field subschemas
+        if model not in cls.models_to_schema_and_fields_dict:
```

Comment:
would this ever happen? I think it should never need to be initted on demand?

Reply:
in the case someone uses this code and forgets to call init() prior to a validate(), this if condition catches that. if/when we want to validate in the indexing flow, i think it would be better to only have to call validate() instead of an init() then validate()